bin/portal_utils/orbit_interface.py

```diff
@@ -0,0 +1,352 @@
+    ERROR33 = "ERROR 33: Missing user's sshpublickey:"
+
+    ALL = [ERROR1, ERROR2, ERROR3, ERROR4, ERROR5, ERROR5, 
+           ERROR6, ERROR7, ERROR8, ERROR9, ERROR10, ERROR11,
+           ERROR12, ERROR17, ERROR20, ERROR21, ERROR22, 
+           ERROR30, ERROR31, ERROR32, ERROR33]
+
+#  Class to manage interactions with ORBIT through 
+#  Delegated Account Management API
+class ORBIT_Interface:
+
+    def __init__(self, base_orbit_url):
+        self._base_orbit_url = base_orbit_url
+        self._DEVNULL = open(os.devnull, 'w')
+
+
+    # Does given string contain one of the defined errors?
+    # If so, return the error, otherwise, return None
+    def find_error_code(self, str):
+        for cd in CODES.ALL:
+            if str.find(cd) >= 0:
+                return cd;
+        return None
+
+    # Return value from CURL on URL
+    def get_curl_output(self, query_url):
+        raw_curl_output = ""
+        try:
+            raw_curl_output = \
+                subprocess.check_output(["curl", "-k", query_url], 
+                                        stderr=self._DEVNULL)
+        except subprocess.CalledProcessError as e:
+            print "Error invoking CURL on %s: Error %s" % \
+                (query_url, e.returncode)
+            sys.exit(e.returncode)
+        return raw_curl_output
+
+    # Invoke an ORBIT REST API call
+    # Create url, make call via curl 
+    #   and exit if an (untolerated) error returned
+    def invoke_orbit_method(self, method, args, 
+                            tolerated_error_codes = []):
+        url = "%s/%s?%s" % (self._base_orbit_url, method, args)
+        output = self.get_curl_output(url)
+        code = self.find_error_code(output)
+        if code and code not in tolerated_error_codes:
+            print "Error in %s call: %s" % (method, output)
+            sys.exit(code)
+    
+
+    # Add ORBIT user to ORBIT group
+    # NOTE: No error trying to add a user to a group to which it already exists
+    def add_user_to_group(self, group_name, user_name):
+        self.invoke_orbit_method("addMemberToGroup", 
+                                 "groupname=%s&username=%s" % \
+                                     (group_name, user_name))
+
+    # Remove ORBIT user to ORBIT group
+    def remove_user_from_group(self, group_name, user_name):
+        self.invoke_orbit_method("removeMemberFromGroup",
+                                 "groupname=%s&username=%s" %  \
+                                     (group_name, user_name))
+
+    # Change ORBIT group admin
+    def change_group_admin(self, group_name, admin_name):
+        self.invoke_orbit_method("changeGroupAdmin", 
+                                 "groupname=%s&username=%s" % \
+                                     (group_name, admin_name))
+
+    # Enable ORBIT user
+    def enable_user(self, user_name):
+        self.invoke_orbit_method("enableUser", 
+                                 "username=%s" % user_name, [CODES.ERROR6])
```

Comment:
How is ERROR6 possible here?

If there is another code that means the user is already enabled, that would be OK here.